src/SocrataSelectQuery.php

```diff
@@ -47,115 +48,51 @@ public function setEndpoint(Endpoint $endpoint = NULL) {
+        $url = $this->endpoint->getMetaDataURL();
+      }
+      else {
+        // Can build the SODA URL now that all the parameters have been set.
+        $url = $this->endpoint->getSodaURL($this->params);
+      }
+      _socrata_dbg($url);
+
+      $res = $client->get($url, [
+        'curl' => $options,
+      ]);
+
+      $retval['headers'] = _socrata_parse_headers($res->getHeaders());
+      if (isset($retval['headers']['x-soda2-fields']) && isset($retval['headers']['x-soda2-types'])) {
+        foreach ($retval['headers']['x-soda2-fields'] as $idx => $name) {
+          $retval['fields'][$name] = $retval['headers']['x-soda2-types'][$idx];
         }
-        _socrata_dbg($curl_url);
-
-        curl_setopt($ch, CURLOPT_URL, $curl_url);
-        $resp = curl_exec($ch);
-        if (FALSE !== $resp) {
-          // Pull info from response and see if we had an error.
-          $info = curl_getinfo($ch);
-          if ($info['http_code'] >= 400) {
-            \Drupal::logger('socrata')->error(
-              'Server returned error code @errno for @url',
-              [
-                '@errno' => $info['http_code'],
-                '@url' => $curl_url,
-              ]
-            );
-
-            break;
-          }
-          else {
-            // Split out response headers into name => value array.
-            list($headers, $data) = explode("\r\n\r\n", $resp, 2);
-            $retval['headers'] = _socrata_parse_headers($headers);
-
-            // Test for redirect in the event curl wasn't able to automagically
-            // follow due to server config.
-            if (!empty($retval['headers']['location'])) {
-              $url = $retval['headers']['location'];
-            }
-            else {
-              // Generate an array mapping fields to types, if provided
-              // Note that the old API returns some fields of type 'meta_data'
-              // (e.g. ':id', ':created_at', ':updated_at').
-              // @todo Add version number for "old".
-              if (isset($retval['headers']['x-soda2-fields']) && isset($retval['headers']['x-soda2-types'])) {
-                foreach ($retval['headers']['x-soda2-fields'] as $idx => $name) {
-                  $retval['fields'][$name] = $retval['headers']['x-soda2-types'][$idx];
-                }
-              }
-
-              // Decode data payload.
-              $retval['data'] = json_decode($data, TRUE);
-            }
-          }
-        }
-        else {
-          \Drupal::logger('socrata')->error(
-            'curl_exec failed: @error [@errno] for @url',
-            [
-              '@error' => curl_error($ch),
-              '@errno' => curl_errno($ch),
-              '@url' => $curl_url,
-            ]
-          );
-
-        }
-      } while (FALSE !== $resp && !empty($retval['headers']['location']));
-
-      // Close and cleanup.
-      curl_close($ch);
+      }
+      $retval['data'] = json_decode($res->getBody()->getContents(), TRUE);
```

Comment:
Let the code breath--add a blank line before this once since the data is separate from the headers.